calculator.rb

```diff
@@ -0,0 +1,66 @@
+#define the operation to use for calculation later 
+def add(a,b)
+    return a + b
+end 
+
+def subtract(a,b)
+    return a - b
+end 
+
+def multiply(a,b)
+    return a * b
+end 
+
+def divide(a,b)
+    return a / b
+end 
+
+
+#introduce the program
+puts "Welcome to the calculator program where you are able to either add, subtract, divide, or multiply two numbers."
+
+#take in the numbers and operation the user would like to perform 
+puts "Input your first number--"
```

Comment:
Nice work making sure the user input is valid! This code is repeated below for the second number -- as such this would be a great opportunity to use a method to DRY up your code.